autoload/phpactor/project/projectRepository.vim

```diff
@@ -0,0 +1,47 @@
+function! phpactor#project#projectRepository#create() abort
+  return {
+        \ 'projects': {},
+        \ 'isEmpty': function('s:isEmpty'),
+        \ 'addProject': function('s:addProject'),
+        \ 'findProjectContainingFile': function('s:findProjectContainingFile'),
+        \ 'listProjects': function('s:listProjects'),
+        \ 'hasProjectWithPrimaryRoot': function('s:hasProjectWithPrimaryRoot'),
+        \ 'hasProject': function('s:hasProject')
+        \ }
+endfunction
+
+function! s:listProjects() dict abort
+  return copy(self.projects)
```

Comment:
I guess the idea was better safe than sorry with the `copy` but we can access it directly through the dictionary.
So could we imagine let the user the care to copy the projects before playing with them in order to avoid making copies every time the method is used (even if it's only to read) ?
And while at it, duplicate with `dict.projects`, choose one of the two if we decide to optimize and remove the `copy`.

Reply:
We can copy projects dictionary within `PhpactorListProjects` implementation instead.

Reply:
I didn't look much at the command to be honest and now that I do I don't get it, the `map()` seems of no use.
And returning only the list of paths doesn't seem that useful.
I would rather remove it for now and wait for a need to emerge before implementing it: [YAGNI](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it)

And even we keep it there is no need to copy it.
The all point of making a copy is to make sure that we don't modify the projects we got internally, but this command only do "read actions".

Reply:
Maybe my need for ensure what projects already exist is more specific for development than normal usage ;)

Reply:
We could also imagine expanding this to "switch" projects (in the future, better to keep this PR small :) )